lib/common/common/src/fixed_length_priority_queue.rs

```diff
@@ -59,20 +58,25 @@ impl<T: Ord> FixedLengthPriorityQueue<T> {
         Some(value.0)
     }
 
-    /// Returns the top N elements sorted ascending.
-    pub fn into_vec(self) -> Vec<T> {
+    /// Consumes the [`FixedLengthPriorityQueue`] and returns a vector
+    /// in sorted (descending) order.
+    pub fn into_sorted_vec(self) -> Vec<T> {
         self.heap
             .into_sorted_vec()
             .into_iter()
             .map(|Reverse(x)| x)
             .collect()
     }
 
-    /// Returns an iterator over the elements in the queue.
-    pub fn iter(&self) -> Iter<'_, T> {
-        Iter {
-            it: self.heap.iter().rev(),
```

Comment:
Also, I don't see a reason why `.rev()` was used.

It would place the smallest element at the end, but that's it; it doesn't fix the order order of all but the smallest element.

I've removed it.